### PR TITLE
Updated links in INSTALL

### PR DESCRIPTION
**Pull request by cimarronm**
Updated dead links for pip and pillow in the INSTALL text

### CODE REVIEW

**Comment by mdboom** [comment]
Thanks.

On a side note -- should we try to run the sphinx linkcheck on Travis to check this kind of stuff?  @jenshnielsen ?

File: INSTALL [hunk 1]
```diff
@@ -243,7 +243,7 @@ ImageMagick
 Optional dependencies
 ^^^^^^^^^^^^^^^^^^^^^
 
-`Pillow <http://python-imaging.github.io/>`__
+`Pillow <https://github.com/python-pillow/Pillow>`__
     If Pillow is installed, matplotlib can read and write a larger
```
Comment (jenshnielsen) [inline INSTALL]: Shouldn't this be http://python-pillow.github.io/ which looks like the homepage rather than the git repo?

**Comment by jenshnielsen** [comment]
It would make sense to me that Travis tests links in INSTALL etc. too.

File: INSTALL (hunk 1 above)
Comment (cimarronm) [inline INSTALL]: Absolutely...will update

**Comment by jenshnielsen** [comment]
IMHO we should not hold this back on improving Travis to catch these issues. I have opened #4205 to remind my self to do that

**Pull request merged**